src/raftstore/store/snap.rs

```diff
@@ -704,14 +710,28 @@ impl Snap {
             } else {
                 let mut key_count = 0;
                 let mut size = 0;
+                let base = if self.limiter.is_some() {
```

Comment:
You can do it like `let limit = self.limiter.map_or(0, |l| l.get_max_bytes_per_time())`.